client/src/components/MainPage/TaskModal/TagInput.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useState, useEffect, useRef } from 'react';
+import axios from 'axios';
+import { HOST } from '../../../constants';
+import styled from 'styled-components';
+import useInput from '../../../hooks/useInput';
+import { RiCloseLine } from 'react-icons/ri';
+import { Tag } from 'GlobalType';
+
+axios.defaults.withCredentials = true; // withCredentials 전역 설정
+
+interface Props {
+  setTagIdx: React.Dispatch<React.SetStateAction<number | undefined>>;
+}
+
+const TagInput = (props: Props) => {
+  const [tagList, setTagList] = useState<Tag[]>([]);
+  const [tagInput, onTagInputChange, setTagInput] = useInput('');
+  const [isTagFocus, setIsTagFocus] = useState<Boolean>(false); // Input창이 Focus될 때만 List 표시
+  const [selectedTag, setSelectedTag] = useState<Tag>(); // 선택된 하나의 태그
+  const [randomColor, setRandomColor] = useState(''); // 새로운 Tag 추가 시 색깔 랜덤 배정
+
+  //TAG GET
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const result = await axios.get(`${HOST}/api/v1/tag`);
+        setTagList(result.data);
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    fetchData();
+  }, [, () => postNewTag]);
+
+  //검색된 Tag
+  const filteredTagList = tagList.filter((itemList) => itemList.title.toUpperCase().includes(tagInput.toUpperCase()));
+
+  //하나의 태그 선택
+  const setTagItem = (e: React.MouseEvent<HTMLDivElement>) => {
+    setSelectedTag(filteredTagList.find((item) => item.idx.toString() === e.currentTarget!.dataset.idx));
+    props.setTagIdx(Number(e.currentTarget.dataset.idx));
+  };
+
+  //태그 선택 해제
+  const unSetTagItem = () => {
+    setSelectedTag(undefined);
```

Comment:
의도적인 값 삭제의 경우 `null`이 더 적절한 것 같습니다.
```suggestion
    setSelectedTag(null);
    props.setTagIdx(null);
```